### PR TITLE
Sp2-98 specify operation point rewrite

### PR DESCRIPTION
**Pull request by frske**
As in the first iteration I did not implemented the operation point correctly, which lead to a major rewrite. 

THERE ARE BREAKING CHANGES IN THIS PR BUT IT WAS NEEDED FROM AN ARCHITECTURAL STANDPOINT

- correctly implements all the requirements for the first scenario (heat only boilers) 
- Sets operation point for the boilers, can calculate net production costs 

The code is probably not the cleanest right now, thats why I am really looking forward to your feedback.

It is also adding a new Model, ResultHolder, which can hold a result computed by the optimizer. 

**Summary of changes which affect other parts of the code**

- modified DataPoints to include a toString override
- Deleted my dummy classes previously used 
- Added production Unit class manually as the pr was not yet merged and modified it a bit, those changes will be added in another pr

I apologize for having forgot yet again to add the jira issue in the beginning of each commit

### CODE REVIEW

**Comment by magwrap** [comment]
Could you please pull the newest changes from dev, and maybe also from SP2-85? I see you merged it before but now it's updated.

**Comment by magwrap** [comment]
Resolved Conflicts

File: Heatington/Optimizer/Optimizer.cs [hunk 1]
```diff
@@ -1,123 +1,210 @@
...
 {
-    private List<ProductionUnitFs> _pUnits = new List<ProductionUnitFs>();
-    private List<FrancescoEnergyData> _energyDataEntries = new List<FrancescoEnergyData>();
-    private List<ResultHolder> _resultEntries = new List<ResultHolder>();
+    private List<DataPoint>? _dataPoints = new List<DataPoint>();
+    private List<ProductionUnit> _productionUnits = new List<ProductionUnit>();
+    public List<ResultHolder>? Results { get; private set; }
 
-    public void OptimizerScenario1()
+    public void LoadData()
```
Comment (pkyeager) [inline Heatington/Optimizer/Optimizer.cs:12]: What do think about using a try-catch statement here?

File: Heatington/Optimizer/Optimizer.cs [hunk 2]
```diff
@@ -1,123 +1,210 @@
...
+                    throw new Exception("WARNING: HEAT DEMAND CAN NOT BE SATISFIED");
+                }
             }
+            return i;
         }
-        return i;
-    }
 
-    // Method which calculates Hourly NetProductionCost for heat only boilers
+        List<ProductionUnit> SelectBoiler(int j)
```
Comment (pkyeager) [inline Heatington/Optimizer/Optimizer.cs:119]: Does this actually "select" the boiler?

File: Heatington/Optimizer/Optimizer.cs [hunk 3]
```diff
@@ -1,123 +1,210 @@
...
+            return;
         }
+        Results.ForEach(Console.WriteLine);
     }
 
-    // Will eventually call AssetManager to get the production units.
-    // Right now it just initialises two and adds them to the list.
-    // SOLID is applied: if I add productionUnits by calling the assetManager it will behave in the same way.
-    private bool GetProductionUnits()
+    private void GetDataPoints()
```
[the remark this answers is not in the document]
Comment (pkyeager) [reply Heatington/Optimizer/Optimizer.cs:161]: PS: I am sorry :-(

**Comment by pkyeager** [comment]
Nice done! You may want to check #12 If you implement the new logic, then I think, there is nothing speaking against the merging.

File: Heatington/Optimizer/Optimizer.cs (hunk 1 above)
Comment (frske) [reply Heatington/Optimizer/Optimizer.cs:12]: I am not sure about that as I think it's the sdm's responsibility.

File: Heatington/Optimizer/Optimizer.cs (hunk 2 above)
Comment (frske) [reply Heatington/Optimizer/Optimizer.cs:119]: Yes, it selects the boiler objects from the production Unit list

**Comment by magwrap** [comment]
conflicts resolved

**Review by rikkiSnaebjornsson — APPROVED** [review]
Looks good

**Pull request merged**